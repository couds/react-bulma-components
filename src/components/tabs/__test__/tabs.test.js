import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from '..';

describe('Tabs component', () => {
  it('Should Exist', () => {
    expect(Tabs).toMatchSnapshot();
  });
  it('Should expose Tab', () => {
    expect(Tabs.Tab).toMatchSnapshot();
  });
  it('Should have Tabs classname', () => {
    const component = renderer.create(<Tabs>Facebook</Tabs>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat Bulma class with classes in props', () => {
    const component = renderer.create(
      <Tabs className="other-class test">Facebook</Tabs>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render as an html section', () => {
    const component = renderer.create(
      <Tabs renderAs="section">This should be a section</Tabs>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have custom inline styles', () => {
    const component = renderer.create(
      <Tabs renderAs="section" style={{ width: 200, zIndex: 1 }}>
        This should be a section with custom styles
      </Tabs>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should accept a react Element as renderAs prop', () => {
    // eslint-disable-next-line react/prop-types
    const Custom = ({ children, ...props }) => (
      <p {...props}>
        Custom
        {children}
      </p>
    );
    const component = renderer.create(
      <Tabs renderAs={Custom}>This should be a p element</Tabs>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render propertly Inside Tabs', () => {
    const component = renderer.create(
      <Tabs>
        <Tabs.Tab>Tab 1</Tabs.Tab>
        <Tabs.Tab>Tab 2</Tabs.Tab>
        <Tabs.Tab>Tab 3</Tabs.Tab>
      </Tabs>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
