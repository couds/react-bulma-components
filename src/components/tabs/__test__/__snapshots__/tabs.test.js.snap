// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component Should Exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "Tab": Object {
    "$$typeof": Symbol(react.forward_ref),
    "defaultProps": Object {
      "active": false,
      "backgroundColor": undefined,
      "children": null,
      "className": "",
      "clearfix": undefined,
      "clipped": undefined,
      "hidden": undefined,
      "invisible": undefined,
      "italic": undefined,
      "marginless": undefined,
      "overlay": undefined,
      "paddingless": undefined,
      "pull": undefined,
      "radiusless": undefined,
      "renderAs": "a",
      "responsive": undefined,
      "shadowless": undefined,
      "style": Object {},
      "textAlignment": undefined,
      "textColor": undefined,
      "textSize": undefined,
      "textTransform": undefined,
      "textWeight": undefined,
      "unselectable": undefined,
    },
    "propTypes": Object {
      "active": [Function],
      "backgroundColor": [Function],
      "children": [Function],
      "className": [Function],
      "clearfix": [Function],
      "clipped": [Function],
      "hidden": [Function],
      "invisible": [Function],
      "italic": [Function],
      "marginless": [Function],
      "overlay": [Function],
      "paddingless": [Function],
      "pull": [Function],
      "radiusless": [Function],
      "renderAs": [Function],
      "responsive": [Function],
      "shadowless": [Function],
      "style": [Function],
      "textAlignment": [Function],
      "textColor": [Function],
      "textSize": [Function],
      "textTransform": [Function],
      "textWeight": [Function],
      "unselectable": [Function],
    },
    "render": [Function],
  },
  "defaultProps": Object {
    "align": null,
    "backgroundColor": undefined,
    "children": null,
    "className": "",
    "clearfix": undefined,
    "clipped": undefined,
    "fullwidth": false,
    "hidden": undefined,
    "invisible": undefined,
    "italic": undefined,
    "marginless": undefined,
    "overlay": undefined,
    "paddingless": undefined,
    "pull": undefined,
    "radiusless": undefined,
    "renderAs": "div",
    "responsive": undefined,
    "shadowless": undefined,
    "size": null,
    "style": Object {},
    "textAlignment": undefined,
    "textColor": undefined,
    "textSize": undefined,
    "textTransform": undefined,
    "textWeight": undefined,
    "type": null,
    "unselectable": undefined,
  },
  "propTypes": Object {
    "align": [Function],
    "backgroundColor": [Function],
    "children": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "fullwidth": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "renderAs": [Function],
    "responsive": [Function],
    "shadowless": [Function],
    "size": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "type": [Function],
    "unselectable": [Function],
  },
  "render": [Function],
}
`;

exports[`Tabs component Should accept a react Element as renderAs prop 1`] = `
<p
  className="tabs"
  style={Object {}}
>
  Custom
  <ul>
    This should be a p element
  </ul>
</p>
`;

exports[`Tabs component Should concat Bulma class with classes in props 1`] = `
<div
  className="tabs other-class test"
  style={Object {}}
>
  <ul>
    Facebook
  </ul>
</div>
`;

exports[`Tabs component Should expose Tab 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "active": false,
    "backgroundColor": undefined,
    "children": null,
    "className": "",
    "clearfix": undefined,
    "clipped": undefined,
    "hidden": undefined,
    "invisible": undefined,
    "italic": undefined,
    "marginless": undefined,
    "overlay": undefined,
    "paddingless": undefined,
    "pull": undefined,
    "radiusless": undefined,
    "renderAs": "a",
    "responsive": undefined,
    "shadowless": undefined,
    "style": Object {},
    "textAlignment": undefined,
    "textColor": undefined,
    "textSize": undefined,
    "textTransform": undefined,
    "textWeight": undefined,
    "unselectable": undefined,
  },
  "propTypes": Object {
    "active": [Function],
    "backgroundColor": [Function],
    "children": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "renderAs": [Function],
    "responsive": [Function],
    "shadowless": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "unselectable": [Function],
  },
  "render": [Function],
}
`;

exports[`Tabs component Should have Tabs classname 1`] = `
<div
  className="tabs"
  style={Object {}}
>
  <ul>
    Facebook
  </ul>
</div>
`;

exports[`Tabs component Should have custom inline styles 1`] = `
<section
  className="tabs"
  style={
    Object {
      "width": 200,
      "zIndex": 1,
    }
  }
>
  <ul>
    This should be a section with custom styles
  </ul>
</section>
`;

exports[`Tabs component Should render as an html section 1`] = `
<section
  className="tabs"
  style={Object {}}
>
  <ul>
    This should be a section
  </ul>
</section>
`;

exports[`Tabs component Should render propertly Inside Tabs 1`] = `
<div
  className="tabs"
  style={Object {}}
>
  <ul>
    <li
      className=""
      style={Object {}}
    >
      <a>
        Tab 1
      </a>
    </li>
    <li
      className=""
      style={Object {}}
    >
      <a>
        Tab 2
      </a>
    </li>
    <li
      className=""
      style={Object {}}
    >
      <a>
        Tab 3
      </a>
    </li>
  </ul>
</div>
`;
